controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +590,101 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS thingy
+			dnsRecord := &infranetworkv1.DNSData{}
+			dnsName := "dns-" + serviceName + fmt.Sprintf("-%d", i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			Log.Info(fmt.Sprintf("Name: %v, Namespace: %v", dnsName, dnsNamespace))
+			test_err := helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsRecord)
+			if test_err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", test_err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsRecord)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", err))
+			}
+
+		}
+	}
+
+	Log.Info("DNS Starts here")
+
+	serviceLabels = map[string]string{
+		common.AppSelector: serviceName,
+	}
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	for i, ovnPod := range podList.Items {
+		if i >= int(*(instance.Spec.Replicas)) {
+			Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+			break
+		}
+		var dnsName string
+		var dnsIP string
+		var hostnames []string
+		dnsName = "dns-" + ovnPod.Name
+		// Get Hostname
+		svc, _ := service.GetServiceWithName(
+			ctx,
+			helper,
+			ovnPod.Name,
+			ovnPod.Namespace,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+		hostnames = append(hostnames, hostname)
+
+		// Get IP
+		net_stat, _ := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+		for _, v := range net_stat {
+			if v.Interface == instance.Spec.NetworkAttachment {
+				dnsIP = v.IPs[0]
```

Comment:
please explain, in a comment, why the first address is the correct one to use (can it be ipv6? is the order guaranteed by API?)

Reply:
It is possible that for the same networkAttachment there are multiple IPs (multiple ipv4 and ipv6). This will depend on the differents ipRanges defined into the NetworkAttachmentDefinition, which at the moment is only [one](https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/gen-netatt.sh#L87-L91). Right now I'm checking the first IPv4 IP inside internalapi network. In case that in the future more IPs needs to be handled we can discuss which would be the best approach.